### PR TITLE
Reduce maximum repr(align(N)) to 2^29

### LINKED ISSUE #49492: Quirks of #repr(align()) errors
2GiB alignments fail:
```
error[E0589]: invalid `repr(align)` attribute: larger than 2147483647
  --> src/main.rs:11:8
   |
11 | #[repr(align(2147483648))]
   |        ^^^^^^^^^^^^^^^^^
```

which could be considered fine, although I don't see why, besides the `Layout::from_size_align` limitation which I think is wrong (https://github.com/rust-lang/rust/issues/32838#issuecomment-368348408). But that's besides the point. The error message is kind of pointless talking about 2GiB - 1:

```
error[E0589]: invalid `repr(align)` attribute: not a power of two
  --> src/main.rs:11:8
   |
11 | #[repr(align(2147483647))]
   |        ^^^^^^^^^^^^^^^^^
```

Because any value larger than 1GiB will actually fail too because they're not powers of two.

Also, The error is unknown on error-index.html: https://doc.rust-lang.org/error-index.html#E0589

### PR DESCRIPTION
**Pull request by varkor**
The current maximum `repr(align(N))` alignment is larger than the maximum alignment accepted by LLVM, which can cause issues for huge values of `N`, as seen in #49492. Fixes #49492.

r? @rkruppe

### CODE REVIEW

File: src/librustc_target/abi/mod.rs [hunk 1]
```diff
@@ -326,9 +326,9 @@ impl AddAssign for Size {
 }
 
 /// Alignment of a type in bytes, both ABI-mandated and preferred.
-/// Each field is a power of two, giving the alignment a maximum value of
-/// 2<sup>(2<sup>8</sup> - 1)</sup>, which is limited by LLVM to a i32,
-/// with a maximum capacity of 2<sup>31</sup> - 1 or 2147483647.
+/// Each field is a power of two, giving the alignment a maximum value
+/// of 2<sup>(2<sup>8</sup> - 1)</sup>, which is limited by LLVM to a
+/// maximum capacity of 2<sup>29</sup> or 536870912.
```
Comment (rkruppe) [inline src/librustc_target/abi/mod.rs]: I'm pretty sure `Align::from_bytes` needs to be updated to reflect this.

**Comment by kennytm** [comment]
(Testing)

r? @rkruppe

File: src/librustc_target/abi/mod.rs (hunk 1 above)
Comment (varkor) [reply src/librustc_target/abi/mod.rs]: Ah, of course!

**Comment by rkruppe** [comment]
cc @eddyb

**Pull request merged**